redskyctl/internal/commands/experiments/experiments.go

```diff
@@ -101,47 +104,91 @@ func (n *name) experimentName() experimentsv1alpha1.ExperimentName {
 	return experimentsv1alpha1.NewExperimentName(n.Name)
 }
 
-// numberSuffixPattern matches the trailing digits, for example the number on the end of a trial name
-var numberSuffixPattern = regexp.MustCompile(`(.*?)(?:[/\-]([[:digit:]]+))?$`)
-
 // parseNames parses a list of arguments into structured names
 func parseNames(args []string) ([]name, error) {
-	var t resourceType
 	names := make([]name, 0, len(args))
+
+	var defaultType string
 	for _, arg := range args {
-		n := name{Type: t, Name: arg, Number: -1}
-		if sm := numberSuffixPattern.FindStringSubmatch(n.Name); sm != nil && sm[2] != "" {
-			n.Number, _ = strconv.ParseInt(sm[2], 10, 64)
-			n.Name = sm[1]
+		argType, argName, argNumber := splitArg(arg, defaultType)
+
+		// Normalize the type, if no name was supplied just update the default type
+		normalType, plural, err := normalizeType(argType)
+		if err != nil {
+			return nil, err
+		}
+		if defaultType == "" && argName == "" && argNumber == "" {
+			defaultType = string(normalType)
+			if plural {
+				// Suspect.
+				defaultType += "s"
+			}
```

Comment:
won't this append an unnecessary s?

```
defaultType = ""
argType = experiments
normalType = experiments

defaultType = experiments
defaultType = experimentss
```

Reply:
The "s" is always added to `normalType` after it is cast to a string and a normal type will never have the "s". The fact that it's not clear is a pretty good indicator of code smell though.

Reply:
I reworked this a little so the "s" gets added in `normalizeType`. I tried going further and adding a `trialNumber` so we could defer getting the trial numbers until later, but the ambiguity between an experiment named `exp-1` and the first trial of the `exp` experiment causes problems when we are deciding if we need to do a list or get. Without injecting server validation into the logic, it's going to be difficult to address this problem 100%.